drivers/platform/aducm3029/trng.c

```diff
@@ -0,0 +1,171 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdlib.h>
+#include <drivers/rng/adi_rng.h>
+#include "trng.h"
+#include "util.h"
+#include "error.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/* LenReload: 0 - 4095 */
+#define TRNG_CNT_VAL    4095
+/* Prescaler: 0 - 10 */
+#define TRNG_PRESCALER  2
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/* Stucture holding the TRNG descriptor. */
+struct trng_desc {
+	/*
+	 * Memory used by the DFP
+	 * At least ADI_RNG_MEMORY_SIZE bytes of 4 bytes aligned memory are
+	 * needed by the DFP driver. The formula is to align memory only.
+	 */
+	uint32_t	dev_mem[(ADI_RNG_MEMORY_SIZE + 3)/4];
+	/* DFP Hanler */
+	ADI_RNG_HANDLE	dev;
+};
+
+/******************************************************************************/
+/**************************** Global Variables ********************************/
+/******************************************************************************/
+
+/** Number of references to the descriptor */
+static uint32_t		nb_references;
+/** TRNG Descriptor */
+static struct trng_desc	g_desc;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/* Initialize the TRNG device and the global descriptor */
+static void gdesc_init()
+{
+	adi_rng_Open(0, g_desc.dev_mem, ADI_RNG_MEMORY_SIZE, &g_desc.dev);
```

Comment:
Since the dev_id is defined in trng_init_param why not use it here?

Reply:
because only device 0 is supported in aducm. dev_id was added to be used by other platforms

Reply:
OK. I would say to use a define then so that it looks better and is more descriptive of what the 0 means.